packages/react-resize-handle/src/hooks/useResizeHandle.ts

```diff
@@ -84,37 +131,47 @@ export const useResizeHandle = (params: UseResizeHandleParams) => {
 
   const currentValue = React.useRef(UNMEASURED);
 
-  const updateElementsAttrs = React.useCallback(() => {
-    const a11yValue = relative
-      ? // If relative mode is enabled, we actually have to measure the element,
-        // because the currentValue is just the px offset.
-        getA11ValueText(elementDimension(elementRef.current, growDirection))
-      : getA11ValueText(currentValue.current);
-
-    const handleAttributes = {
-      role: 'slider',
-      'aria-valuemin': minValue,
-      ...(maxValue < Number.MAX_SAFE_INTEGER && { 'aria-valuemax': maxValue }),
-      ...(a11yValue && { 'aria-valuetext': a11yValue }),
-      'aria-orientation':
-        growDirection === 'end' || growDirection === 'start'
-          ? 'horizontal'
-          : 'vertical',
-    };
-
-    Object.entries(handleAttributes).forEach(([key, value]) => {
-      handleRef.current?.setAttribute(key, String(value));
-    });
-
-    // Make sure to only apply the value if it's not the initial value!
-    if (currentValue.current !== UNMEASURED) {
-      wrapperRef.current?.style.setProperty(
-        variableName,
-        `${currentValue.current}px`
-      );
-      onChange?.(currentValue.current);
-    }
-  }, [getA11ValueText, maxValue, minValue, onChange, variableName]);
+  const updateElementsAttrs = React.useCallback(
+    (event?: NativeTouchOrMouseEvent | KeyboardEvent) => {
+      const a11yValue = relative
+        ? // If relative mode is enabled, we actually have to measure the element,
+          // because the currentValue is just the px offset.
+          getA11ValueText(elementDimension(elementRef.current, growDirection))
+        : getA11ValueText(currentValue.current);
+
+      const handleAttributes = {
+        role: 'slider',
+        'aria-valuemin': minValue,
+        ...(maxValue < Number.MAX_SAFE_INTEGER && {
+          'aria-valuemax': maxValue,
+        }),
+        ...(a11yValue && { 'aria-valuetext': a11yValue }),
+        'aria-orientation':
+          growDirection === 'end' || growDirection === 'start'
+            ? 'horizontal'
+            : 'vertical',
+      };
+
+      Object.entries(handleAttributes).forEach(([key, value]) => {
+        handleRef.current?.setAttribute(key, String(value));
+      });
+
+      // Make sure to only apply the value if it's not the initial value!
+      if (currentValue.current !== UNMEASURED) {
+        wrapperRef.current?.style.setProperty(
+          variableName,
+          `${currentValue.current}px`
+        );
+
+        const eventToPass = event || new CustomEvent(EVENTS.setValue);
+        onChange?.(
+          eventToPass,
+          getUpdateEventData(eventToPass, currentValue.current)
```

Comment:
```suggestion
        onChange?.(
          eventToPass,
          getUpdateEventData(event, currentValue.current)
```

Should `getUpdateEventData()` create event instead?

Reply:
that won't work, as `onChange` is an `EventHandler` and it expects the first parameter to be of type `Event`, which is also not nullable

Reply:
Did not get it. Why it cannot be following?

```ts
const getUpdateEventData = (
  event: NativeTouchOrMouseEvent | KeyboardEvent | null,
  value: number
): ResizeHandleUpdateEventData => {
  // ...
    default:
      return {
        type: EVENTS.setValue,
        value,
        event: new CustomEvent(EVENTS.setValue),
      };
}
```